From fbbe8987b7da681f7e9b2a582c1951b9ae3b6552 Mon Sep 17 00:00:00 2001
From: Thomas White <tomwhite@apache.org>
Date: Mon, 8 Oct 2012 11:44:42 +0000
Subject: [PATCH 1203/1357] MR1: MAPREDUCE-4706. FairScheduler#dump(): Computing of # running maps and reduces is commented out. Contributed by Karthik Kambatla.

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1395520 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 31c02cb4b0be9f298f11f7fac584ee603afb4973)

Reason: Bug fix (Better debugging of FS)
Ref: CDH-8210
Author: Karthik Kambatla
(cherry picked from commit df1cad6a6ec42818840282816ddad9fb137bbe32)
---
 .../org/apache/hadoop/mapred/FairScheduler.java    |   12 ++----------
 1 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/src/contrib/fairscheduler/src/java/org/apache/hadoop/mapred/FairScheduler.java b/src/contrib/fairscheduler/src/java/org/apache/hadoop/mapred/FairScheduler.java
index 1b90631..604a4ac 100644
--- a/src/contrib/fairscheduler/src/java/org/apache/hadoop/mapred/FairScheduler.java
+++ b/src/contrib/fairscheduler/src/java/org/apache/hadoop/mapred/FairScheduler.java
@@ -1018,16 +1018,8 @@ public class FairScheduler extends TaskScheduler {
           else return p1.getName().compareTo(p2.getName());
         }});
       for (Pool pool: pools) {
-        int runningMaps = 0;
-        int runningReduces = 0;
-        for (JobInProgress job: pool.getJobs()) {
-          JobInfo info = infos.get(job);
-          if (info != null) {
-            // TODO: Fix
-            //runningMaps += info.runningMaps;
-            //runningReduces += info.runningReduces;
-          }
-        }
+        int runningMaps = pool.getMapSchedulable().getRunningTasks();
+        int runningReduces = pool.getReduceSchedulable().getRunningTasks();
         String name = pool.getName();
         eventLog.log("POOL",
             name, poolMgr.getPoolWeight(name), pool.getJobs().size(),
-- 
1.7.0.4

